File: backend/mongo/src/main/kotlin/se/filmstund/database/mongo/entities/Ticket.kt
package se.filmstund.database.mongo.entities

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import se.filmstund.domain.id.GoogleId
import java.time.LocalDate
import java.time.LocalTime
import java.util.*

@Document
@Deprecated(message = "")
internal data class Ticket(
  @Id
  val id: String = "",
  val showingId: UUID,
  val assignedToUser: GoogleId,
  val profileId: String?,
  val barcode: String,
  val customerType: String,
  val customerTypeDefinition: String,
  val cinema: String,
  val cinemaCity: String?,
  val screen: String,
  val seat: Seat,
  val date: LocalDate,
  val time: LocalTime,
  val movieName: String,
  val movieRating: String, // 15 år, 11 år etc.
  val showAttributes: List<String> // "textad", "en" etc
)

@Deprecated(message = "")
data class Seat(val row: Int, val number: Int)

File: web/src/use-cases/showing-tickets/__mocks__/ticket.js
export const createMockSeat = (row, number, x, y) => ({
  row,
  number,
  coordinates: {
    x,
    y
  },
  dimensions: {
    width: 10,
    height: 10
  }
});

export const createMockTicket = (id, row, number) => ({
  id,
  barcode: "this-is-barcode",
  customerType: "mock-customer-type",
  customerTypeDefinition: "mock-customer-type-definition",
  cinema: "mock-cinema",
  screen: "mock-screen",
  profileId: "mock-profile-id",
  seat: {
    row,
    number
  },
  date: "2018-10-10",
  time: "20:00",
  movieName: "mock-movie-name",
  movieRating: "R",
  showAttributes: ["3D"]
});
